src/composables/passwordToggleFunction.js

```diff
@@ -0,0 +1,92 @@
+"use client";
+import { useState } from "react";
+
+const PasswordToggle = () => {
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+  const [inputValue, setInputValue] = useState("");
+  const [isDefaultPassword, setIsDefaultPassword] = useState(true);
+
+  const handleInputChange = (e) => {
+    const value = e.target.value;
+    setInputValue(value);
+    setIsDefaultPassword(value === "" || !isPasswordVisible);
+  };
+
+  const handleToggleClick = () => {
+    if (inputValue !== "") {
+      setIsPasswordVisible(!isPasswordVisible);
+      setIsDefaultPassword(false);
+    } else {
+      setIsPasswordVisible(false);
+      setIsDefaultPassword(true);
+    }
+  };
+
+  const eyeIcon = isPasswordVisible ? (
```

Comment:
Put this in a useEffect hook

And it should watch fo change in `isPassword` `[isPasswordVisible]`